gen_unigram crashes due to unaligned access (Bus error) on sparc64

Trying to build `libpinyin` on sparc64 on Linux fails with:

```
rm -f phrase_index.bin pinyin_index.bin addon_phrase_index.bin addon_pinyin_index.bin bigram.db gb_char.bin gbk_char.bin opengram.bin merged.bin art.bin culture.bin economy.bin geology.bin history.bin life.bin nature.bin people.bin science.bin society.bin sport.bin technology.bin
../utils/storage/gen_binary_files --table-dir ../data
../utils/storage/import_interpolation --table-dir ../data < ../data/interpolation2.text
../utils/training/gen_unigram --table-dir ../data
make[3]: *** [Makefile:555: bigram.db] Bus error
make[3]: *** Deleting file 'bigram.db'
make[3]: Leaving directory '/build/reproducible-path/libpinyin-2.8.1/data'
make[2]: *** [Makefile:487: all-recursive] Error 1
make[2]: Leaving directory '/build/reproducible-path/libpinyin-2.8.1'
make[1]: *** [Makefile:392: all] Error 2
make[1]: Leaving directory '/build/reproducible-path/libpinyin-2.8.1'
```

Full build log in: https://buildd.debian.org/status/fetch.php?pkg=libpinyin&arch=sparc64&ver=2.8.1-1&stamp=1760835176&raw=0

This is due to an unaligned access in `utils/training/gen_unigram` as GDB shows:

```
Core was generated by `../utils/training/gen_unigram --table-dir ../data'.
Program terminated with signal SIGUSR1, User defined signal 1.
#0  0x000001000004ac90 in pinyin::SubPhraseIndex::get_range (this=0x1000022d910, range=...) at ./src/storage/phrase_index.cpp:649
649         for (poffset = end; poffset > begin + 1; --poffset) {
(gdb) bt
#0  0x000001000004ac90 in pinyin::SubPhraseIndex::get_range (this=0x1000022d910, range=...) at ./src/storage/phrase_index.cpp:649
#1  pinyin::FacadePhraseIndex::get_range (this=<optimized out>, phrase_index=1 '\001', range=...) at ./src/storage/phrase_index.cpp:627
#2  0x0000010000046394 in generate_unigram (phrase_files=0x7feffeea9e8) at ./utils/training/gen_unigram.cpp:63
#3  0x000001000001d898 in main (argc=3, argv=<optimized out>) at ./utils/training/gen_unigram.cpp:106
```

Access to a Linux sparc64 machine for testing can be obtained through the GCC Compile Farm, see: https://gcc.gnu.org/wiki/CompileFarm

--- a/src/storage/phrase_index.cpp
+++ b/src/storage/phrase_index.cpp
@@ -647,7 +647,7 @@ int SubPhraseIndex::get_range(/* out */ PhraseIndexRange & range){
     /* remove trailing zeros. */
     const table_offset_t * poffset = NULL;
     for (poffset = end; poffset > begin + 1; --poffset) {
-        if (0 !=  *(poffset - 1))
+        if (0 !=  UnalignedMemory<table_offset_t>::load(poffset - 1))
             break;
     }
 

